vita3k/gui/src/content_manager.cpp

```diff
@@ -29,6 +31,19 @@
 #include <util/safe_time.h>
 
 namespace gui {
+namespace {
+template <typename T>
+auto get_recursive_directory_size(const T &path) {
+    const auto &path_list = fs::recursive_directory_iterator(path);
+    const auto pred = [](const auto acc, const auto &app) {
+        if (fs::is_regular_file(app.path()))
+            return acc + fs::file_size(app.path());
+        return acc;
+    };
+    return boost::accumulate(path_list, 0ul, pred);
```

Comment:
Increasing dependency on Boost when `std::accumulate` already exists is redundant. Are there any advantages over using the `std` implementation?

Reply:
For this function & case, there is one advantage / difference:
* `fs::recursive_directory_iterator` is accepted as a "first-class citizen" datatype and the special `fs` iterators are accepted "as is", just like `std` iterators.
* In the `std` version, you must use special `fs::{begin, end}()` free functions for `fs::recursive_directory_iterator` to work.

In this case, the `boost` version is very streamlined & minimalistic to use, making it less error-prone and understandable.